### PR TITLE
[6.3.0] compute resource profile

### PR DESCRIPTION
**Pull request by ldjebran**
Issue:  https://github.com/SatelliteQE/robottelo/issues/4076

Bugs:  
affect compute resource EC2: https://bugzilla.redhat.com/show_bug.cgi?id=1451626
affect compute resource RHEV profile: https://bugzilla.redhat.com/show_bug.cgi?id=1452534

Work done:
- UI Fixes
- Infrastructure to support Compute resource Profile
- Overall 6 tests implemented 
- 2 qe_test_coverage for the above bugs

Compute resource profile forms support implemented with this PR: 
- EC2
- RHEV  

```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test -v tests/foreman/ui/test_computeresource_ec2.py
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, cov-2.4.0
collected 15 items 
2017-05-19 12:40:58 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_negative_add_image_ec2_with_invalid_name <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_negative_create_ec2_with_invalid_name <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_access_ec2_with_custom_profile <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_access_ec2_with_default_profile <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_add_image_ec2_with_name <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_create_ec2_with_description <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_create_ec2_with_name <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_delete_ec2 <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_provision_ec2_host_with_image <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_provision_ec2_with_compute_profile <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_provision_ec2_with_custom_compute_settings <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_provision_ec2_with_host_group <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_retrieve_ec2_vm_list <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_update_ec2_name <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_ec2.py::Ec2ComputeResourceTestCase::test_positive_update_ec2_organization <- robottelo/decorators/__init__.py SKIPPED

======================================== 3 passed, 12 skipped in 150.02 seconds ========================================
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ 
```

```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_computeresource_rhev.py -v -k "test_positive_access_rhev_with_custom"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, cov-2.4.0
collected 18 items 
2017-05-19 12:45:23 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_custom_profile <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_custom_profile_with_template <- robottelo/decorators/__init__.py SKIPPED

================================================= 16 tests deselected ==================================================
================================= 1 passed, 1 skipped, 16 deselected in 60.77 seconds ==================================
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ 
```

### CODE REVIEW

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 1]
```diff
@@ -465,8 +465,98 @@ def test_positive_access_rhev_with_custom_profile(self):
...
+                    ),
+                    dict(
+                        name='nic2',
+                        network='zstream_el7'
+                    ),
+                ],
+                storage=[
+                    dict(
+                        size='10',
+                        storage_domain='NFS-BOS',
```
Comment (san7ket) [inline tests/foreman/ui/test_computeresource_rhev.py]: storage_domain should be fetched from robotello.properties file

File: robottelo.properties.sample [hunk 2]
```diff
@@ -303,6 +303,31 @@ ssh_key=
 # 'semanage port -a -t websm_port_t -p tcp <port-range>'
 port_range=9091, 14999
 
+# [ec2]
+# AWS EC2 to be added as a compute resource.
+# Access key ID of the IAM user
+# access_key=
```
Comment (san7ket) [inline robottelo.properties.sample]: Some of these settings are not supposed to be taken from properties files directly as we don't have a shared account currently, but should be sourced from Ci . @kbidarkar What do you think?

File: robottelo.properties.sample [hunk 3]
```diff
@@ -303,6 +303,31 @@ ssh_key=
...
+# Secret access key of the IAM user
+# secret_key=
```
Comment (san7ket) [inline robottelo.properties.sample]: Same as above

File: tests/foreman/ui/test_computeresource_rhev.py (hunk 1 above)
Comment (ldjebran) [inline tests/foreman/ui/test_computeresource_rhev.py]: @san7ket , thanks for your review, storage_domain correspond to which field in robotello.properties file  as config data for RHEV exist but this field is missing

Comment (san7ket) [inline tests/foreman/ui/test_computeresource_rhev.py]: At the time of adding RHEV settings, only tier1 was automated , so we didn't need the storage_domain field . But now, maybe  we could add the new field

File: tests/foreman/ui/test_computeresource_rhev.py [hunk 4]
```diff
@@ -465,8 +465,98 @@ def test_positive_access_rhev_with_custom_profile(self):
...
+            )
+            self.compute_resource.set_profile_values(
+                name, COMPUTE_PROFILE_LARGE,
+                cluster=self.rhev_datacenter,
+                cores=2,
+                memory=1024,
+                network_interfaces=[
+                    dict(
+                        name='nic1',
+                        network='zstream_el6'
```
Comment (san7ket) [inline tests/foreman/ui/test_computeresource_rhev.py]: I am not sure about this, hard-coding the VLANS doesn't seem a good idea , as this automation could be run any boxes which will not necessarily have these VLANS. Don't know how we could handle this?
@kbidarkar @rochacbruno any thoughts on this

File: tests/foreman/ui/test_computeresource_rhev.py (hunk 1 above)
Comment (ldjebran) [inline tests/foreman/ui/test_computeresource_rhev.py]: @san7ket , thanks will update the properties file and settings with this properties (and accordingly the test)

File: tests/foreman/ui/test_computeresource_rhev.py (hunk 4 above)
Comment (san7ket) [inline tests/foreman/ui/test_computeresource_rhev.py]: This value is already present in robotello.properties

```
# [vlan_networking]
# subnet=SUBNET_VALUE
# netmask=NETMASK_VALUE
# bridge=BRIDGE_VALUE
# gateway=GATEWAY_VALUE
```

**Comment by ldjebran** [comment]
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ py.test tests/foreman/ui/test_computeresource_rhev.py -v -k "test_positive_access_rhev_with_custom or test_positive_delete_rhev or test_positive_access_rhev_with_default_profile"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, cov-2.4.0
collected 18 items 
2017-05-22 11:46:54 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_custom_profile <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_custom_profile_with_template <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_default_profile <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_delete_rhev <- robottelo/decorators/__init__.py PASSED

================================================= 14 tests deselected ==================================================
================================= 3 passed, 1 skipped, 14 deselected in 184.81 seconds =================================
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ 
```

File: robottelo.properties.sample (hunk 2 above)
Comment (kbidarkar) [inline robottelo.properties.sample]: @san7ket , good point, these values can come in from CI, but now the point is these values should be masked, now we need to be sure , whether these masked values can be passed in here. I feel may be not. Need to figure a way to get past this.

**Comment by ldjebran** [comment]
```console
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$  py.test tests/foreman/ui/test_computeresource_rhev.py -v -k "test_positive_access_rhev_with_custom or test_positive_delete_rhev or test_positive_access_rhev_with_default_profile"
================================================= test session starts ==================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.32, pluggy-0.4.0 -- /home/dlezz/.pyenv/versions/sat-6.3.0/bin/python2.7
cachedir: .cache
rootdir: /home/dlezz/projects/robottelo-fork, inifile:
plugins: xdist-1.15.0, services-1.2.1, mock-1.6.0, cov-2.4.0
collected 18 items 
2017-05-25 11:34:44 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_custom_profile <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_custom_profile_with_template <- robottelo/decorators/__init__.py SKIPPED
tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_access_rhev_with_default_profile <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_computeresource_rhev.py::RhevComputeResourceTestCase::test_positive_delete_rhev <- robottelo/decorators/__init__.py PASSED

================================================= 14 tests deselected ==================================================
================================= 3 passed, 1 skipped, 14 deselected in 174.10 seconds =================================
(sat-6.3.0) dlezz@elysion:~/projects/robottelo-fork$ 
```

**Comment by ldjebran** [comment]
@san7ket @kbidarkar san7ket I updated with all the comments regarding RHEV, still question about EC2, 
@sghai , @omaciel, @rochacbruno  what do you think? 
or we need to discuss about in the weekly meeting?

**Comment by san7ket** [comment]
For handling the EC2 Credentials, we need to figure out a different way. Apart from that, ACK

**Comment by rochacbruno** [comment]
There are some recommendations on boto lib docs http://boto3.readthedocs.io/en/latest/guide/configuration.html

File: robottelo.properties.sample (hunk 2 above)
Comment (kbidarkar) [inline robottelo.properties.sample]: @san7ket , @ldjebran and @rochacbruno , adding support for passing the `access_key` and `secret_key` in the robottelo.properties file in the `robottelo-ci` project.

**Comment by kbidarkar** [comment]
Once, the access_key and secret_key PR in robottelo-ci is also raised then we can merge both of these PR's together.

**Comment by kbidarkar** [comment]
We have 2 ACK's merging this PR.

**Pull request merged**